contrib/cli/great_expectations_contrib/package.py

```diff
@@ -0,0 +1,229 @@
+import importlib
+import inspect
+import json
+import logging
+import os
+import sys
+from dataclasses import asdict, dataclass
+from enum import Enum
+from typing import Any, ClassVar, Dict, List, Optional, Type
+
+from great_expectations.expectations.expectation import Expectation
+
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+# Type alias that will need to be updated to reflect the complex nature of the 'run_diagnostics' return object
+Diagnostics = Dict[str, Any]
+
+
+@dataclass(frozen=True)
+class ExpectationCompletenessCheck:
+    message: str
+    passed: bool
+
+
+@dataclass(frozen=True)
+class ExpectationCompletenessChecklist:
+    experimental: List[ExpectationCompletenessCheck]
+    beta: List[ExpectationCompletenessCheck]
+    production: List[ExpectationCompletenessCheck]
+
+
+@dataclass(frozen=True)
+class PackageCompletenessStatus:
+    concept_only: int
+    experimental: int
+    beta: int
+    production: int
+    total: int
+
+
+@dataclass(frozen=True)
+class RenderedExpectationDetail:
+    name: str
+    tags: List[str]
+    supported: List[str]
+    status: ExpectationCompletenessChecklist
+
+
+@dataclass(frozen=True)
+class Dependency:
+    text: str
+    link: str
+    version: Optional[str]
+
+
+@dataclass(frozen=True)
+class GitHubUser:
+    username: str
+    full_name: Optional[str]
+
+
+class SocialLinkType(Enum):
+    TWITTER = "TWITTER"
+    INSTAGRAM = "INSTAGRAM"
+    LINKEDIN = "LINKEDIN"
+    MEDIUM = "MEDIUM"
+
+
+@dataclass(frozen=True)
+class SocialLink:
+    account_type: SocialLinkType
+    identifier: str
+
+
+@dataclass(frozen=True)
+class DomainExpert:
+    full_name: str
+    social_links: List[SocialLink]
+    picture: str
+
+
+class Maturity(Enum):
+    CONCEPT_ONLY = "CONCEPT_ONLY"
+    EXPERIMENTAL = "EXPERIMENTAL"
+    BETA = "BETA"
+    PRODUCTION = "PRODUCTION"
+
+
+@dataclass(frozen=True)
+class GreatExpectationsContribPackage:
+    # Core
+    package_name: Optional[str] = None
+    icon: Optional[str] = None
+    description: Optional[str] = None
+    expectations: Optional[List[RenderedExpectationDetail]] = None
+    expectation_count: Optional[int] = None
+    dependencies: Optional[List[Dependency]] = None
+    maturity: Optional[Maturity] = None
+    status: Optional[PackageCompletenessStatus] = None
+
+    # Users
+    owners: Optional[List[GitHubUser]] = None
+    contributors: Optional[List[GitHubUser]] = None
+    domain_experts: Optional[List[DomainExpert]] = None
+
+    # Metadata
+    version: Optional[str] = None
+
+    # Existing configuration
+    config_file: ClassVar[str] = ".great_expectations_package.json"
+
+    def _to_json_file(self) -> None:
+        json_dict = self._to_json_dict()
+        data = json.dumps(json_dict, indent=4)
+        with open(self.config_file, "w") as f:
```

Comment:
We tried to implement dataclasses in some Mario Pod features but had to revert to old patterns due to some unexpected behavior - that being said, our ultimate goal is to definitely re-enable dataclasses down the line.

I'm okay either way here - since the vast majority of these classes are simple bags of data, I think dataclasses make a lot of sense. However, I also see a lot of value in consistency around the codebase (even if this isn't a true part of GE).

Happy to defer to @donaldheppner here.